lib/link_thumbnailer/scrapers/opengraph/image.rb

```diff
@@ -20,7 +21,11 @@ def value
           end
 
           def model
-            nodes.map { |n| modelize(n, n.attributes['content'].to_s) }
+            nodes.map do |n|
+              if LinkThumbnailer::URI.new(n.attributes['content']).valid?
```

Comment:
I can refactor the line to : 
```
nodes.to_a.keep_if { |n| LinkThumbnailer::URI.new(n.attributes['content']).valid? }.map { |n| modelize(n, n.attributes['content'].to_s) }
```

OR

```
nodes.to_a.keep_if do |n|
  LinkThumbnailer::URI.new(n.attributes['content']).valid?
end.map do |n|
  modelize(n, n.attributes['content'].to_s)
end
```

I prefer the second one, which one would you prefer?